pkg/policy/mapstate.go

```diff
@@ -16,40 +16,55 @@ package policy
 
 import (
 	"github.com/cilium/cilium/pkg/identity"
-	"github.com/cilium/cilium/pkg/maps/policymap"
 	"github.com/cilium/cilium/pkg/option"
 	"github.com/cilium/cilium/pkg/policy/trafficdirection"
 )
 
 var (
 	// localHostKey represents an ingress L3 allow from the local host.
-	localHostKey = policymap.PolicyKey{
+	localHostKey = Key{
 		Identity:         identity.ReservedIdentityHost.Uint32(),
 		TrafficDirection: trafficdirection.Ingress.Uint8(),
 	}
 
 	// worldKey represents an ingress L3 allow from the world.
-	worldKey = policymap.PolicyKey{
+	worldKey = Key{
 		Identity:         identity.ReservedIdentityWorld.Uint32(),
 		TrafficDirection: trafficdirection.Ingress.Uint8(),
 	}
 )
 
 // MapState is a state of a policy map.
-type MapState map[policymap.PolicyKey]MapStateEntry
+type MapState map[Key]MapStateEntry
 
-// MapStateEntry is the configuration associated with a PolicyKey in a
+// Key is the userspace representation of a policy key in BPF. It is
```

Comment:
Ok, we are getting somewhere. If this is supposed to be (B), then it should be in a different package. Here we are already mixing both A and C. There should be a package that can import both A and C so it can perform the translation from A to B and from B to C.

`[ (A) User Policy ] -> [ (B) Endpoint Policy ] -> [ (C) Datapath Policy ]`

The fields in this key structure are already datapath aware they have no relation with this policy package.

Reply:
@aanm this is meant to be an iterative change. I agree that we need to get to B, but I want to do this in a piecemeal, easy-to-review way. The reason I am doing this is so I can encapsulate what currently is translated policy for the endpoint in a single type, which I am doing in 
https://github.com/cilium/cilium/pull/6270 (which is based on top of this PR). In order to undo the current way we do things in policy translation, I am of the belief that if we can decouple the endpoint from policy translation (which is what this PR seeks to do), then we can get to B. After #6270 is reviewed / merged, I can then work to decouple what the generated policy for an endpoint is from the datapath implementation. But, to do that, I need to do some of the translation bits within the policy package, which for now, will include this `MapState` type.